app/modules/detectx/detectx_model.php

```diff
@@ -0,0 +1,67 @@
+<?php
+class Detectx_model extends \Model {
+
+	function __construct($serial='')
+	{
+		parent::__construct('id', 'detectx'); //primary key, tablename
+		$this->rs['id'] = '';
+		$this->rs['serial_number'] = $serial;
```

Comment:
Please add a UNIQUE rt statement: `$this->rt['serial_number'] = 'VARCHAR(255) UNIQUE';`